src/cosmos-keys.ts

```diff
@@ -49,16 +49,21 @@ export function getSeed(randomBytesFunc: (size: number) => Buffer = randomBytes)
   return mnemonic
 }
 
-export function getNewWallet(randomBytesFunc: (size: number) => Buffer = randomBytes): Wallet {
+export function getNewWallet(randomBytesFunc: (size: number) => Buffer = randomBytes, networkPrefix: string): Wallet {
   const mnemonic = getSeed(randomBytesFunc)
-  return getNewWalletFromSeed(mnemonic)
+  return getNewWalletFromSeed(mnemonic, networkPrefix)
 }
 
 // NOTE: this only works with a compressed public key (33 bytes)
-export function getCosmosAddress(publicKey: Buffer): string {
+export function getCosmosAddress(publicKey: Buffer, networkPrefix: string): string {
   const message = CryptoJS.enc.Hex.parse(publicKey.toString('hex'))
   const address = CryptoJS.RIPEMD160(CryptoJS.SHA256(message) as any).toString()
-  const cosmosAddress = bech32ify(address, `cosmos`)
+  if (networkPrefix === `regen`) {
```

Comment:
let's not handle the network specific logic in here. `networkPrefix` can be `bech32Prefix`. The integrating application then can take car of defining where to get the prefix from.